src/components/space/ChartsHeader.js

```diff
@@ -1,40 +1,68 @@
-import React from 'react';
+import React, { useContext } from 'react';
+import { useLocation } from 'react-router-dom';
 import { Tooltip, Typography } from '@material-ui/core';
 import { Info } from '@material-ui/icons';
 import Grid from '@material-ui/core/Grid';
 import { makeStyles } from '@material-ui/core/styles';
 import { useTranslation } from 'react-i18next';
+import { SpaceDataContext } from '../../contexts/SpaceDataProvider';
 
 const useStyles = makeStyles((theme) => ({
-  cell: {
+  spaceName: {
+    fontWeight: 900,
+  },
+  leftCell: {
+    display: 'flex',
+    justifyContent: 'flex-start',
+    alignItems: 'center',
+  },
+  rightCell: {
     display: 'flex',
+    justifyContent: 'flex-end',
+    alignItems: 'center',
   },
   root: {
-    marginBottom: theme.spacing(2),
     flexGrow: 1,
-    padding: theme.spacing(2),
+    paddingTop: theme.spacing(2),
+    paddingRight: theme.spacing(2),
+    paddingLeft: theme.spacing(2),
+    paddingBottom: theme.spacing(1),
+  },
+  alert: {
+    display: 'flex',
+    alignItems: 'center',
+    marginTop: theme.spacing(1),
   },
 }));
 
 function ChartsHeader() {
   const { t } = useTranslation();
   const classes = useStyles();
+  const { spaceName } = useContext(SpaceDataContext);
+  const location = useLocation();
+
+  if (location.pathname.slice(0, 9) === '/embedded') {
```

Comment:
You should get the path from the Router. It's more elegant.

Reply:
Phew. This one surprisingly took some time. I used a solution that combines useLocation and matchPath. More elegant than slicing, which you're right is not cool. If you don't like this let me know if you have other ideas.